core/modules/startup/rootwidget.js

```diff
@@ -25,6 +25,9 @@ exports.startup = function() {
 	$tw.rootWidget.addEventListener("tm-modal",function(event) {
 		$tw.modal.display(event.param,{variables: event.paramObject, event: event});
 	});
+	$tw.rootWidget.addEventListener("tm-show-switcher",function(event) {
+		$tw.modal.display("$:/core/ui/LayoutSwitcher",{variables: event.paramObject, event: event});
```

Comment:
@BurningTreeC event.paramObject is a hashmap of variables to the modal. So if tm-show-switcher is called with a variable indicating switcher type, that gets passed to the modal and you can use the variable in the caption field of the modal tiddler to get it to display in the modal title.

Reply:
Thanks @saqimtiaz - that's an idea. But should that title change when we change the tabs to say "Theme"? Should it then show "Theme Switcher"?

Reply:
@BurningTreeC I guess yes. Though to be honest, having just taken a fresh look at the demo I am not so sure the title of the modal and the tabs needs to match, otherwise it is just redundant. 

@Jermolene perhaps take a fresh look at the demo. It may be more appropriate to come up with a more generalized name for the modal since the tabs themselves are called after their respective switchers.

Reply:
@Jermolene @BurningTreeC bumping this as it would be good to have for 5.1.23

Reply:
@saqimtiaz I just don't know how to proceed